File: event_app/models/event.ts
import mongoose from "mongoose";

const Schema = mongoose.Schema;
const eventSchema = new Schema(
  {
    name: {
      type: String,
      required: true,
      unique: true
    },
    description: {
      type: String,
    },
    startDateTime: {
      type: Date,
      required: true,
    },
    endDateTime: {
      type: Date,
      required: true,
    },
    venue: {
      type: String,
      required: true
    },
    ticketsAvailable: {
      type: Number,
      required: true
    },
    publish: {
      type: Boolean,
      default: false
    }
  },
  {
    timestamps: true
  }
)

const Event = mongoose.models.Event || mongoose.model('Event', eventSchema);

export default Event;

File: booking_server/internal/events/event.go
package events

import (
	"time"
)

type Event struct {
	ID               string    `json:"_id"`
	Name             string    `json:"name"`
	Description      string    `json:"description"`
	StartDateTime    time.Time `json:"startDateTime"`
	EndDateTime      time.Time `json:"endDateTime"`
	Venue            string    `json:"venue"`
	TicketsAvailable int64     `json:"ticketsAvailable"`
	Publish          bool      `json:"publish"`
}
